tools/bootnode/bootnode.go

```diff
@@ -88,10 +84,11 @@ func extractPrivateKey() *ecdsa.PrivateKey {
 		}
 
 	} else {
-		privKey, err = ecdsa.GenerateKey(ECDSACurve, rand.Reader)
+		privInterfaceKey, _, err := crypto.GenerateSecp256k1Key(rand.Reader)
```

Comment:
do x509.ParseECPrivateKey(b) supports Secp256k1Key?

Reply:
this does, I can add a test for this code path

Reply:
great

Reply:
Thanks for catching that! turns out it doesn't parse correctly, so I unmarshalled it using another one from the libp2p crypto library.